app.rb

```diff
@@ -0,0 +1,120 @@
+require_relative 'book'
+require_relative 'person'
+require_relative 'rental'
+
+class Student < Person
+  def initialize(id, name, age, parent_permission)
+    super(id, name, age)
+    @parent_permission = parent_permission
+  end
+end
+
+class App
+  def initialize
+    @books = []
+    @people = []
+    @rentals = []
+  end
+
+  def list_books
+    puts "\nBooks:"
+    @books.each do |book|
+      puts "Title: #{book.title}, Author: #{book.author}"
+    end
+  end
+
+  def list_people
+    puts "\nPeople:"
+    @people.each do |person|
+      puts "ID: #{person.id}, Name: #{person.name}, Age: #{person.age}"
+    end
+  end
+
+  def create_person
+    print 'Enter the name of the person: '
+    name = gets.chomp
+
+    print 'Enter the age of the person: '
+    age = gets.chomp.to_i
+
+    print 'Does the person have parent permission? (y/n): '
+    parent_permission = gets.chomp.downcase == 'y'
+
+    print 'Is the person a student or a teacher? (student/teacher): '
+    type = gets.chomp.downcase
+
+    if type == 'student'
+      create_student(name, age, parent_permission)
+    elsif type == 'teacher'
+      create_teacher(name, age, parent_permission)
+    else
+      puts 'Invalid person type. Person not created.'
+    end
+  end
+
+  def create_student(name, age, parent_permission)
+    student = Student.new(@people.length + 1, name, age, parent_permission)
+    @people.push(student)
+    puts 'Student created successfully.'
+  end
+
```

Comment:
Fantastic job adding most of the  required changes. 👏🏾 

There's still an issue with creating a student.

![image](https://github.com/Ochiengsteven/school-library/assets/38088818/ee0d6757-70c9-47fe-819d-e7c0c3b5801e)


### Observations:

[OPTIONAL] I noticed that you created a new student class inside the `app.rb` file. this approach is affecting flow of the application because it eliminates the requires associations like the student's relationship with classroom. It would be best to required the student class that already exist in the application.

### Suggestions

Kindly note that the errors are mostly `ArgumentError`s. this means that there's an issue with the way your parameters are setup for the relevant classes.

- The `person.rb` file: The error suggests that the the `initialize` method on line 9 expects 0 arguments. This could be because the `name` and `age` parameters are already given value `name = 'Unknown', age = 0`. I recommend that the values are removed so that name and age parameters can be accepted.
- Also I attempted removing the new student class in the `app.rb` file and required the `student` file and encountered the same issue. An `ArgumentError` caused by the assigned values for `age` and `name`.  Should you decide to restore using the Student class in `student.rb` you can fix the parameters of the class to resolve the issues.